flake.nix

```diff
@@ -21,73 +21,111 @@
     ...
   }: let
     lib = nixpkgs.lib;
+    ncl = nci.lib.nci-lib;
     mkRootPath = rel:
       builtins.path {
         path = "${toString ./.}/${rel}";
         name = rel;
       };
+    filteredSource = let
+      pathsToIgnore = [
+        ".envrc"
+        ".ignore"
+        ".github"
+        "runtime"
+        "screenshot.png"
+        "book"
+        "contrib"
+        "docs"
+        "README.md"
+        "CHANGELOG.md"
+        "shell.nix"
+        "default.nix"
+        "grammars.nix"
+        "flake.nix"
+        "flake.lock"
+      ];
+      ignorePaths = path: type: let
+        split = lib.splitString "/" path;
+        actual = lib.drop 4 split;
```

Comment:
This seems specific - is this assuming the depth of the helix repo on the building machine?

Reply:
That is for removing `/nix/hash-source/` part after splitting the nix store path:
```
nix-repl> lib.splitString "/" "/nix/store/m2wybrc3grfn76hxcv4gyiyncng17z30-hello-2.12.1"
[ "" "nix" "store" "m2wybrc3grfn76hxcv4gyiyncng17z30-hello-2.12.1" ]
```
I can add a comment there. I did not want to use `nix-filter` here since IMO it's to complicated to just pull another dep in, we don't need all of the stuff in there for filtering the helix source.